src/services/fetch.service.ts

```diff
@@ -4,24 +4,18 @@ class FetchService {
   private async fetchData(endpoint: string, init?: RequestInit): Promise<Response> {
     const url = this.getUrl(endpoint);
 
-    if (this.responseMap.has(url)) {
-      return await this.responseMap.get(url)!;
-    }
+    if (this.responseMap.has(url)) return this.responseMap.get(url)!;
 
     const fetchPromise = fetch(url, init);
 
     this.responseMap.set(url, fetchPromise);
```

Comment:
ich denke wir müssen den response in dem map speichern nur wenn response ok ist...

Reply:
Dafür müsste der promisse resolved werden bevor er gespeichert wird. Was dann wiederum dazu führt das mehermals der gleiche request gestartet werden kann. 

ich könnte alternativ bei einem fehler den gespeicherten promise löschen so haben wir nur responses die resolved sind in der map